source/CardGames.py

```diff
@@ -235,23 +260,169 @@ def printHands(self, players: dict):
 
 
 class GoFish:
-  def __init__(self, players: "list[Player]"):
+  def __init__(self):
     self.playerinfo = []
-    self.players = players
-    self.value_wanted = int
-    self.suit_wanted = ""
-    self.player_asked = ""
-    self.player_turn = ""
-    self.info_player_turn = int
-    self.info_player_asked = int
+    self.players = []
+    self.player_dict = {}
+    self.value_wanted = 0
+    self.player_asked = None
+    self.player_turn = None
+    self.info_player_turn = 0
+    self.info_player_asked = 0
+    self.deck = Deck()
+    self.dealer = Dealer(self.deck)
+    self.width = os.get_terminal_size().columns
 
   def start_game(self):
-    deck = Deck()
-    dealer = Dealer(deck)
-    dealer.dealCards(5, self.players)
+    
+    print("*._."*(self.width//4))
+    print()
+    print("Welcome to Go Fish!".center(self.width))
+    print()
+    print("*._."*(self.width//4))
+
+    player_number = self.multiplayer_info()
+    self.point_tracker_interface(player_number)
+
+    
+    self.dealer.dealCards(5, self.players)
     for player in self.players:
+      print("\n",player)
       player.showHand(player.hand)
+
+    self.game_loop()
+
```

Comment:
wow this is way fancier now, Great job Keren

Reply:
Thank you! I didn't add the rules, but I think Robert did that so we can combine them. Thanks for the feedback.